libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/RouteExpirationHandler.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.navigation.base.internal.route
+
+import androidx.annotation.MainThread
+import androidx.annotation.VisibleForTesting
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.utils.internal.Time
+
+@MainThread
+object RouteExpirationHandler {
+
+    private val routesExpirationTimes = mutableMapOf<String, Long>()
+
+    fun updateRouteExpirationData(route: NavigationRoute, refreshTtl: Int?) {
+        if (refreshTtl != null) {
+            routesExpirationTimes[route.id] = Time.SystemClockImpl.seconds() + refreshTtl
+        }
+    }
+
+    fun isRouteExpired(route: NavigationRoute): Boolean {
+        val routeExpirationTime = routesExpirationTimes[route.id]
+        return routeExpirationTime != null && routeExpirationTime <= Time.SystemClockImpl.seconds()
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    internal fun clear() {
```

Comment:
Why there is no cleanup outside of the tests?

Reply:
I didn't notice any mechanism to cleanup ttls from old routes. For example when user does `mapboxNavigaiton.setRoutes(emptyList())` or in case of reroute. Do you cleanup this data somehow?

Reply:
No, why? Their expiration times did not change. And nothing prevents the user from setting this route again.

Reply:
Current approach creates very small memory leak. It leaks ~1 MB for every 50k routes that get through our system. It wouldn't be a concern for mobile apps that are often killed, but as far as I'm aware, some automotive customers don't restart navigation apps for months